src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,180 @@
+package duke.parser;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import duke.command.AddDeadlineCommand;
+import duke.command.AddEventCommand;
+import duke.command.AddTodoCommand;
+import duke.command.ClearCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.data.exception.DukeException;
+
+/**
+ * Represents the parser of user-given commands.
+ */
+public class Parser {
+
+    private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
+    private static final Pattern TASK_INDEX_ARGS_FORMAT = Pattern.compile("(?<taskIndex>-?\\d+)");
+    private static final Pattern TODO_DATA_ARGS_FORMAT = Pattern.compile("(?<description>[^/]+)");
+    private static final Pattern DEADLINE_DATA_ARGS_FORMAT =
+            Pattern.compile("(?<description>[^/]+)"
+                    + " /by (?<by>[^/]+)");
+    private static final Pattern EVENT_DATA_ARGS_FORMAT =
+            Pattern.compile("(?<description>[^/]+)"
+                    + " /from (?<from>[^/]+)"
+                    + " /to (?<to>[^/]+)");
+    private static final Pattern KEYWORD_DATA_ARGS_FORMAT = Pattern.compile("(?<keyword>\\w+)");
```

Comment:
Nice clean and concise way to save the different formats. I think I will try and use this in my code